### PR TITLE
Add documentation around pallet coupling

### PR DESCRIPTION
**Pull request by kianenigma**
substrate.io deprecation companion: https://github.com/substrate-developer-hub/substrate-docs/pull/2139
pba-content companion: https://github.com/Polkadot-Blockchain-Academy/pba-content/pull/978

partially inspired by: https://github.com/paritytech/polkadot-sdk/issues/3535

### CODE REVIEW

File: docs/sdk/src/reference_docs/frame_pallet_coupling.rs [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+//!
+//! This reference document explains how FRAME pallets can be combined to interact together.
+//!
+//! It is suggested to re-read [`crate::polkadot_sdk::frame_runtime`], notably the information
+//! around [`frame::pallet_macros::config`]. Recall that:
+//!
+//! > The configuration trait of a pallet. Mandatory. Allows a pallet to receive types at a later
+//! > point from the runtime that wishes to contain it. It allows the pallet to be parameterized
+//! > over both types and values.
+//!
```
Comment (Ank4n) [inline docs/sdk/src/reference_docs/frame_pallet_coupling.rs:11]: I would reword this a bit to fit the story here. Also, wondering if we should use the word dependency injection which is much more common programming term.
```suggestion
//! > Configuration trait of a pallet: It allows a pallet to receive types at a later
//! > point from the runtime that wishes to contain it. It allows the pallet to be parameterized
//! > over both types and values.
//!
```

File: docs/sdk/src/reference_docs/frame_pallet_coupling.rs [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+//! ## Example
+//!
+//! Consider the following example, in which `pallet-foo` needs another pallet to provide the block
+//! author to it, and `pallet-author` which has access to this information.
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", pallet_foo)]
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", pallet_author)]
+//!
+//! ### Tight Coupling Pallets
+//!
+//! To tightly couple `pallet-foo` and `pallet-author`, we use a Rust's super-trait system. When a
```
Comment (Ank4n) [inline docs/sdk/src/reference_docs/frame_pallet_coupling.rs:53]:
```suggestion
//! To tightly couple `pallet-foo` and `pallet-author`, we use Rust's supertrait system. When a
```

File: docs/sdk/src/reference_docs/frame_pallet_coupling.rs [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+//! from:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", loose_usage)]
+//!
+//! Then, if `pallet-author` implements this glue-trait:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", pallet_author_provider)]
+//!
+//! And upon the creation of the runtime, the two pallets are linked together as such:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", runtime_author_provider)]
+//!
+//! Crucially, in when using loose coupling, we gain the flexibility of providing different
```
Comment (Ank4n) [inline docs/sdk/src/reference_docs/frame_pallet_coupling.rs:90]:
```suggestion
//! Crucially, when using loose coupling, we gain the flexibility of providing different
```

File: docs/sdk/src/reference_docs/frame_pallet_coupling.rs [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+//!
+//! For example, all pallets in `polkadot-sdk` that needed to work with currencies could have been
+//! tightly coupled with [`pallet_balances`]. But, `polkadot-sdk` also provides [`pallet_assets`]
+//! (and more implementations by the community), therefore all pallets use traits to loosely couple
+//! with balances or assets pallet. More on this in [`crate::reference_docs::frame_currency`].
+//!
+//! ## Further References
+//!
+//! <https://www.youtube.com/watch?v=0eNGZpNkJk4>
+//! <https://substrate.stackexchange.com/questions/922/pallet-loose-couplingtight-coupling-and-missing-traits>
```
Comment (Ank4n) [inline docs/sdk/src/reference_docs/frame_pallet_coupling.rs:125]: Otherwise it renders both items on the same line
```suggestion
//! - <https://www.youtube.com/watch?v=0eNGZpNkJk4>
//! - <https://substrate.stackexchange.com/questions/922/pallet-loose-couplingtight-coupling-and-missing-traits>
```

**Review by Ank4n — APPROVED** [review]

File: docs/sdk/src/reference_docs/frame_pallet_coupling.rs [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", loose_usage)]
+//!
+//! Then, if `pallet-author` implements this glue-trait:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", pallet_author_provider)]
+//!
+//! And upon the creation of the runtime, the two pallets are linked together as such:
+#![doc = docify::embed!("./src/reference_docs/frame_pallet_coupling.rs", runtime_author_provider)]
+//!
+//! Crucially, when using loose coupling, we gain the flexibility of providing different
+//! implementations of `AuthorProvider`, such that different users of a `pallet-foo` can use
```
Comment (muharem) [inline docs/sdk/src/reference_docs/frame_pallet_coupling.rs:91]: or providing no-op implementation, very common in our domain

**Review by muharem — APPROVED** [review]

File: docs/sdk/src/reference_docs/frame_pallet_coupling.rs (hunk 5 above)
Comment (kianenigma) [reply docs/sdk/src/reference_docs/frame_pallet_coupling.rs:111]: Done.
